op-node/rollup/derive/channel.go

```diff
@@ -151,17 +158,44 @@ func (ch *Channel) Reader() io.Reader {
 // The L1Inclusion block is also provided at creation time.
 // Warning: the batch reader can read every batch-type.
 // The caller of the batch-reader should filter the results.
-func BatchReader(r io.Reader, maxRLPBytesPerChannel uint64) (func() (*BatchData, error), error) {
-	// Setup decompressor stage + RLP reader
-	zr, err := zlib.NewReader(r)
+func BatchReader(r io.Reader, maxRLPBytesPerChannel uint64, isFjord bool) (func() (*BatchData, error), error) {
```

Comment:
What if we took a peekable/discard reader instead of a generic reader here? Realistically the underlying type is a bytes.Buffer

Also, what if we passed the chainspec object to the batcher reader rather than flags like the limit + isFjord

Reply:
I was thinking the same, instead of passing on more and more individual config settings to just pass the spec + timestamp (or a new _chain rules_ struct, like in geth). However, I didn't want to bloat the PR more, so thought it's fine this time. We can do this in a follow up, or next time we require even more protocol context in the `BatchReader`.

Reply:
Thanks, created a [ticket](https://github.com/ethereum-optimism/optimism/issues/10522)